components/tikv_util/src/metrics/threads_linux.rs

```diff
@@ -287,6 +289,116 @@ lazy_static! {
     };
 }
 
+#[inline]
+fn get_name(command: &str) -> String {
+    if command != "" {
+        return command.to_owned();
+    }
+    String::from("anony")
+}
+
+/// Use to collect cpu usages and disk I/O rates
+pub struct ThreadInfoStatistics {
+    pid: pid_t,
+    last_instant: Instant,
+    cpu_times: HashMap<i32, f64>,
+    read_ios: HashMap<i32, u64>,
+    write_ios: HashMap<i32, u64>,
+}
+
+impl ThreadInfoStatistics {
+    pub fn new() -> Self {
+        let pid = unsafe { libc::getpid() };
+
+        let mut cpu_times = HashMap::default();
+        let mut read_ios = HashMap::default();
+        let mut write_ios = HashMap::default();
+        let tids = get_thread_ids(pid).unwrap();
+        for tid in tids {
+            if let Ok(stat) = pid::stat_task(pid, tid) {
```

Comment:
@overvenus 

seem we have already had a place to collect threads' CPU?

Reply:
The existing "ThreadsCollector" collects the total cpu times and total bytes of disk io for prometheus.
The new struct "ThreadInfoStatistics" is used for collecting the rates of cpu times and disk io, wrapping the "ThreadsCollector" also seems complicated.

Reply:
The `ThreadsCollector` collects per-thread statistics, and it indeed reports to Prometheus.

-----

How about letting `ThreadsCollector` wraps `ThreadInfoStatistics ` so there is no duplicate code?

Reply:
I tried to let ThreadsCollector to wrap ThreadInfoStatistics, but ThreadsCollector uses three different types (CounterVec, IntGaugeVec, IntCounterVec) to record the threads' statistics, it seems different to unify the collecting procedure, (I'm not sure whether they can be simplied by using template function), and wrapping also may induce extra overhead for the ThreadsCollector.